app/services/doc_auth/mock/result_response.rb

```diff
@@ -34,20 +36,27 @@ def errors
             image_metrics = file_data.dig('image_metrics')
             failed = file_data.dig('failed_alerts')
             passed = file_data.dig('passed_alerts')
-            liveness_result = file_data.dig('liveness_result')
+            face_match_result = file_data.dig('portrait_match_results', 'face_match_result')
             classification_info = file_data.dig('classification_info')
             # Pass and doc type is ok
-            if [doc_auth_result, image_metrics, failed, passed,
-                liveness_result, classification_info].any?(&:present?)
+            if [doc_auth_result, image_metrics, failed, passed, face_match_result,
+                classification_info].any?(&:present?)
               # Error generator is not to be called when it's not failure
               # allows us to test successful results
-              return {} if doc_auth_result == 'Passed' && id_type_supported?
+              if @selfie_check_performed
+                return {} if
+                  doc_auth_result == 'Passed' &&
+                  face_match_result == 'Pass' &&
+                  id_type_supported?
+              elsif doc_auth_result == 'Passed' && id_type_supported?
+                return {}
+              end
```

Comment:
I am guessing syntactically there might be a better way to do this. I thought about extracting lines 48-50 into a private method. Unsure if that's the best way to do this though.

Reply:
One liner or another function.
```ruby
doc_auth_result == 'Passed' && id_type_supported && (@selfie_check_performed ? face_match_result == 'Pass' : true)
```